pkgs/by-name/fi/fish/package.nix

```diff
@@ -317,6 +318,7 @@ stdenv.mkDerivation (finalAttrs: {
   preConfigure = ''
     patchShebangs ./build_tools/git_version_gen.sh
     patchShebangs ./tests/test_driver.py
+    substituteInPlace ./tests/test_driver.py --replace-warn '"python3",' '"${testPython}/bin/python3",'
```

Comment:
Why is this required on Darwin and not Linux?

Reply:
On macOS the build runs unsandboxed, so python3 resolves to /usr/bin/python3 instead of the nixpkgs one; pointing to `${testPython}/bin/python3` ensures we always use the interpreter that has `pexpect`.

Reply:
But patching shebangs in `./tests/test_driver.py` should point it to the nixpkgs python, no? I think something else is going on. Are `nativeCheckInputs` available in `preConfigure` (for `patchShebangs` to pick up)?

Maybe moving python to `nativeBuildInputs` would fix it. Or moving test shebang patching to the check phase.